definitions/npm/mkdirp_v0.5.x/flow_>=v0.25.x/mkdirp_v0.5.x.js

```diff
@@ -0,0 +1,36 @@
+declare module 'mkdirp' {
+
+  declare type FsImplementation = {
+    stat: (
+      path: string,
+      cb: (error: typeof Error, made: ?string) => void
+    ) => void,
+    mkdir: (
+      path: string,
+      mode: number,
+      cb: (error: typeof Error, made: ?string) => void
+    ) => void
+  }
+
+  declare type FsImplementationSync = {
+    statSync: (path: string) => mixed,
+    mkdirSync: (path: string, mode: number) => void
+  }
+
+  declare function sync(
+    path: string,
+    options?: {
+      mode?: string,
+      fs?: FsImplementationSync,
+    }
+  ): ?string;
+
+  declare module.exports: (
+    path: string,
+    options?: {
+      mode?: string,
+      fs?: FsImplementation,
+    } | (error: typeof Error, made: ?string) => void,
```

Comment:
Is this an ok way to annotate something that supports both `path, callback` and `path, options, callback`?